### PR TITLE
[common] create DeviceTypes

### PR DESCRIPTION
**Pull request by michelececcacci**

### CODE REVIEW

File: src/main/java/homer/common/DeviceTypesImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    private final Set<String> validTypes;
+    private final Set<String> startingTypes = new LinkedHashSet<>(
+        List.of("LIGHT", "OUTLET", "COUTLET", "LOUTLET", "A/C", "HEATER")
+    );
+
+    /**
+     * Maybe we could pass startingTypes as a parameter here.
+     */
+    public DeviceTypesImpl() {
```
Comment (aleemont1) [inline src/main/java/homer/common/DeviceTypesImpl.java:21]:
```suggestion
    public DeviceTypesImpl(final Set<> startingTypes) {
```

Or whatever startingTypes should be.

**Review by aleemont1 — APPROVED** [review]
LGTM

File: src/main/java/homer/common/DeviceTypesImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+    Pre-populated valid device types.
+ */
+public final class DeviceTypesImpl implements DeviceTypes {
+
+    private final Set<String> validTypes;
+    private final Set<String> startingTypes = new LinkedHashSet<>(
+        List.of("LIGHT", "OUTLET", "COUTLET", "LOUTLET", "A/C", "HEATER")
+    );
```
Comment (smagnaniuni) [inline src/main/java/homer/common/DeviceTypesImpl.java:16]: How are we supposed to retrieve the types from a collection when we want to use it?

Comment (smagnaniuni) [reply src/main/java/homer/common/DeviceTypesImpl.java:16]: @aleemont1 @michelececcacci maybe we should scrap all of this and just use retrieve the class names via reflection if we need to display them.
Same if we want to use configuration files, we will use the class names to identify the devices.

Comment (michelececcacci) [reply src/main/java/homer/common/DeviceTypesImpl.java:16]: > leave DeviceInfo with just the id in there.
Feels like the better choice. By the way, how do we plan on assigning the id to the devices?

Comment (smagnaniuni) [reply src/main/java/homer/common/DeviceTypesImpl.java:16]: Tbh I would just create the ID inside the DeviceInfoImpl, we don't have to edit or do anything to it.

Comment (michelececcacci) [reply src/main/java/homer/common/DeviceTypesImpl.java:16]: wait, i actually meant `scrap DeviceInfo too and just keep the id` feels like the best option. sorry for the confusion

**Pull request merged**